gax-grpc/src/main/java/com/google/longrunning/OperationsClient.java

```diff
@@ -138,17 +142,28 @@ protected OperationsClient(OperationsSettings settings) throws IOException {
     this.executor = channelAndExecutor.getExecutor();
     this.channel = channelAndExecutor.getChannel();
 
+    CredentialsProvider credentialsProvider = settings.getCredentialsProvider();
+    CallCredentials callCredentials = null;
+    if (credentialsProvider.getCredentials() != null) {
+      callCredentials = MoreCallCredentials.from(credentialsProvider.getCredentials());
+    }
+    ClientInitContext clientContext =
+        ClientInitContext.newBuilder()
+            .setExecutor(this.executor)
+            .setChannel(this.channel)
+            .setCallCredentials(callCredentials)
+            .build();
```

Comment:
I agree this makes things nicer. It causes a little weirdness however. To maintain compatibility with `UnaryCallable`, it needs to accept `Channel`, not `ManagedChannel`. But then the generated client now cannot call `channel.shutdown()` because `Channel` doesn't have the method.

I made a design around the problem. Please let me know what you think.